service_provider.go

```diff
@@ -1210,6 +1210,10 @@ func (sp *ServiceProvider) MakeLogoutRequest(idpURL, nameID string) (*LogoutRequ
 			SPNameQualifier: sp.Metadata().EntityID,
 		},
 	}
+	if sessionIndex != "" {
+		req.SessionIndex = &SessionIndex{sessionIndex}
+	}
```

Comment:
Can we get rid of the conditional statement? Can we get the same results if `req.SessionIndex` is `nil` vs `""`?

```suggestion
	req.SessionIndex = &SessionIndex{sessionIndex}
```

Reply:
I enclosed it in the if statement, because `req.SessionIndex` is a pointer and if that's `nil` then the xml marshaller skips the field. If I would instantiate SessionIndex regardless of the `sessionIndex` value then the SessionIndex would be marshalled to xml with an empty value in case of empty `sessionIndex` and that would change the current behaviour.